src/core/regimes.rkt

```diff
@@ -272,21 +266,25 @@
   ;; d for previous index
   ;; This is where the high level bulk of the algorithm is applied
   ;; We get the final splitpoints by applying add-splitpoints as many times as we want
-  (define-values (pa pb pc pd) (initial))
-  (define-values (fa fb fc fd)
-    (let loop ([pa pa] [pb pb] [pc pc] [pd pd])
-      (define-values (na nb nc nd) (add-splitpoint pa pb pc pd))
+  (define-values (pa pb pd) (initial))
+  (define-values (fa fb fd)
+    ; short circuit if there is no other alts to consider
+    (if (> num-candidates 1)
+      (let loop ([pa pa] [pb pb] [pd pd])
+      (define-values (na nb nd) (add-splitpoint pa pb pd))
       (if (equal? nb pb) ;; only need to compare candidate index
-          (values na nb nc nd)
-          (loop na nb nc nd))))
+          (values na nb nd)
+          (loop na nb nd)))
+    (values pa pb pd)))
     
     ;; From here down is messy code translating from 4 vectors back to
     ;; the original list of split points
     (define fixed-final (make-vector num-points))
+    (define fp (list->vector (range 1 (+ num-points 1))))
     (for ([idx (in-range 0 num-points)])
       (define a (flvector-ref fa idx))
       (define b (vector-ref fb idx))
-      (define c (vector-ref fc idx))
+      (define c (vector-ref fp idx))
```

Comment:
Isn’t c just idx + 1?

Reply:
Ya I will clean this up on the next pass.